### DISCUSSION TITLE
Change Volume Output of Cefsharp.BrowserSubprocess

### DISCUSSION THREAD

**Discussion by Noobicle** [cefsharp/CefSharp#4625] [Q&A] [answered]
Hello,

Is there something I can call to adjust the volume levels of the hosted browser?

I can see it in Windows Mixer listed as Cefsharp.BrowserSubprocess but I can't find how to control the volume of it.

My only solution so far is use the windows API to adjust volume but when I search for processes with the name Cefsharp.BrowserSubprocess I of course end up with 6 references and I'm unsure which PID to use. Also a good chance other programs will be using Cefsharp.BrowserSubprocess and I don't want to interfere with their outputs.

Best regards.

**Comment by amaitland** [comment, marked as answer]
> Is there something I can call to adjust the volume levels of the hosted browser?

There's no API for that currently. Only the option to mute/un-mute. A set volume API would  need to be added `upstream`.

See https://github.com/cefsharp/CefSharp/wiki/General-Usage for background.

> My only solution so far is use the windows API to adjust volume but when I search for processes with the name Cefsharp.BrowserSubprocess I of course end up with 6 references and I'm unsure which PID to use.

`Chromium` uses a separate sub process for audio by default, this should be identifiable with the args `--type=utility --utility-sub-type=audio.mojom.AudioService`. Look in Task Manager at the command line args and you'll see that `Chrome` etc will have the same.

> Also a good chance other programs will be using Cefsharp.BrowserSubprocess and I don't want to interfere with their outputs.

You'll have your own copy of the exe within your applications path, so you can identify it via it's path. 

Alternatively you can self host the sub process (use your own app exe), see https://github.com/cefsharp/CefSharp/wiki/SelfHost-BrowserSubProcess

**Comment by Noobicle** [comment]
> > My only solution so far is use the windows API to adjust volume but when I search for processes with the name Cefsharp.BrowserSubprocess I of course end up with 6 references and I'm unsure which PID to use.
> 
> `Chromium` uses a separate sub process for audio by default, this should be identifiable with the args `--type=utility --utility-sub-type=audio.mojom.AudioService`. Look in Task Manager at the command line args and you'll see that `Chrome` etc will have the same.
> 


Perfect, thank you very much for the args hint that will be suitable :).

Thank you so much!
